src/middlewared/middlewared/utils/__init__.py

```diff
@@ -381,3 +402,65 @@ def load_classes(module, base, blacklist):
                     classes.append(attr)
 
     return classes
+
+
+class LoadPluginsMixin(object):
+
+    def __init__(self, overlay_dirs):
+        self.overlay_dirs = overlay_dirs or []
+        self._schemas = Schemas()
+        self._services = {}
+        self._services_aliases = {}
+
+    def _load_plugins(self, on_module_begin=None, on_module_end=None, on_modules_loaded=None):
+        from middlewared.service import Service, CRUDService, ConfigService, SystemServiceService
```

Comment:
Same here

Reply:
I didnt change that, just moved code. I think it was because circular dependency